packages/kotti-single-select/src/SingleSelect.vue

```diff
@@ -68,8 +74,24 @@ export default {
 			queryString: '',
 			selectedLabel: '',
 			visible: false,
+			formOptions: null,
+			formOptionsContainer: null,
+			input: null,
+			formOptionStyle: '',
 		}
 	},
+	mounted() {
+		this.formOptions = this.$refs['formOptions']
+		this.input = this.$refs['input']
```

Comment:
@tomquinonero  this will result in memory leaks as the ref-ed elements can be changed when doing partial rerendering.

the element you have appended to the body and refer to in the variables would not reflect the elements that are rendered by vue.

I would recommend you watch the $refs.formOptions and $refs.input and make sure to run destroy and recreate logic when they change, seeing that you're using v-show instead of v-if would I guess hide this bug though, so perhaps it is not as critical but is never the less present.

Reply:
you can 
```
watch() {
  '$res.formOptions'(formOptions) {
    this.formOptions = formOptions
    if (!formOptions) {
       // remove the element from body like in the destroy hook
    } else {
      // get select option DOMNode in body, if it's present replace it with new one, if it's not present create it.
    }
  }
}
```

note that you'll need to add an element to capture the events being bubbled (like click) and funnel it back into the select component, otherwise the click events are leaking outside in the body and not being captured by the select (hence why I recommended using vue-portal)